internals/secrethub/generate.go

```diff
@@ -60,17 +62,24 @@ func NewGenerateSecretCommand(io ui.IO, newClient newClientFunc) *GenerateSecret
 
 // Register registers the command, arguments and flags on the provided Registerer.
 func (cmd *GenerateSecretCommand) Register(r command.Registerer) {
-	clause := r.Command("generate", "Generate a random secret.")
-	clause.Arg("secret-path", "The path to write the generated secret to").Required().PlaceHolder(secretPathPlaceHolder).StringVar(&cmd.firstArg)
-	clause.Flag("length", "The length of the generated secret. Defaults to "+strconv.Itoa(defaultLength)).PlaceHolder(strconv.Itoa(defaultLength)).Short('l').SetValue(&cmd.lengthFlag)
-	clause.Flag("min", "<charset>:<n> Ensure that the resulting password contains at least n characters from the given character set. Note that adding constraints reduces the strength of the secret. When possible, avoid any constraints.").SetValue(&cmd.mins)
-	clause.Flag("clip", "Copy the generated value to the clipboard. The clipboard is automatically cleared after "+units.HumanDuration(cmd.clearClipboardAfter)+".").Short('c').BoolVar(&cmd.copyToClipboard)
-	clause.Flag("charset", "Define the set of characters to randomly generate a password from. Options are all, alphanumeric, numeric, lowercase, uppercase, letters, symbols and human-readable. Multiple character sets can be combined by supplying them in a comma separated list. Defaults to alphanumeric.").Default("alphanumeric").HintOptions("all", "alphanumeric", "numeric", "lowercase", "uppercase", "letters", "symbols", "human-readable").SetValue(&cmd.charsetFlag)
-	clause.Flag("symbols", "Include symbols in secret.").Short('s').Hidden().SetValue(&cmd.symbolsFlag)
-	clause.Arg("rand-command", "").Hidden().StringVar(&cmd.secondArg)
-	clause.Arg("length", "").Hidden().SetValue(&cmd.lengthArg)
-
-	command.BindAction(clause, cmd.Run)
+	clause := r.CreateCommand("generate", "Generate a random secret.")
+	clause.Args = cobra.RangeArgs(1, 3)
+	//clause.Arg("secret-path", "The path to write the generated secret to").Required().PlaceHolder(secretPathPlaceHolder).StringVar(&cmd.firstArg)
```

Comment:
This argument should be registered

Reply:
But I agree, all these comments containing the old way of registering arguments should be removed.

Reply:
I think for in-depth review, the comments show what arguments were required for the specific command. I kept them just commented to ensure that we don't lose functionality. Once we get the green light (the transition is good), we can remove them